### ISSUE TITLE
Generated setup.rb in --standlone mode should interpolate platform at runtime

### ISSUE THREAD

**Issue by sschuberth** [bundler/bundler#4591]
In `--standalone` mode, for native extensions ([rugged](https://github.com/libgit2/rugged) in this example) the generated `setup.rb` contains a line like

    $:.unshift "#{path}/../#{ruby_engine}/#{ruby_version}/extensions/x64-mingw32/2.3.0/rugged-0.25.0b2"

IMO the `x64-mingw32` platform part should not be hard-coded here, but interpolated at runtime just like `#{ruby_engine}` and `#{ruby_version}`. Doing so would be a step in the right direction to actually support native extensions for multiple platforms in `--standalone` mode.

Using Bundler 1.12.3 with Ruby 2.3 on Windows.

**Comment by sschuberth** [comment]
To clarify, the literal `x64-mingw32` should be replaced with `#{Gem::Platform::local}`.

**Comment by segiddins** [comment]
@sschuberth a PR doing just that would be much appreciated!

**Comment by sschuberth** [comment]
@segiddins I was try to do so before I filed this issue, but I was unable to find the correct place where the extensions-specific string is constructed :-(

**Comment by segiddins** [comment]
Ah so the issue is that technically the standalone script doesn't require rubygems :/

**Comment by sschuberth** [comment]
So, are you saying it also *must* not be made to require rubygems? If so, can't we just use `#{RbConfig::CONFIG['arch']}` instead of `#{Gem::Platform::local}`?

**Comment by segiddins** [comment]
No, the arch and the gem platform string are almost always different.

@indirect thoughts on this?

**Comment by colby-swandale** [comment]
ping @indirect

**Comment by indirect** [comment]
I think it would be ideal if we could avoid hard-coding the platform, but I think we should be able to build a standalone file that works without requiring rubygems by using the list of platforms in the lockfile, right? The standalone file can never be used on a platform until after `install --standalone` has been run, I think we can use that to generate a correct file without needing rubygems at runtime.

**Comment by sschuberth** [comment]
On a somewhat related note, in the path shouldn't the `2.3.0` after `x64-mingw32` also be replaced with `#{ruby_version}`?
